### PR TITLE
Open subvendors in a sheet

### PR DESCRIPTION
**Pull request by robojumper**
As suggested on the Discord

grafik
grafik

### CODE REVIEW

**Comment by bhollis** [comment]
I added a `zIndex` parameter to ItemPopup to allow for this to work from Armory, but the global item popup doesn't pass it through. You could add that, or switch to handling popups the way Armory does (look at its ItemGrid component).

**Comment by robojumper** [comment]
Okay so I tried that, and after some fiddling with the z-indices, now the problem is that the global item popup doesn't exist in the sheet hierarchy, so it the sheet's `ClickOutsideRoot` cannot dismiss the item. Not sure if there's an easy way to switch it to the armory approach -- the popup trigger is deep in `VendorItems->VendorItemComponent->VendorItemDisplay`...

**Comment by bhollis** [comment]
Yup that's the problem. For vendors we can probably switch to the simpler item grid (I've been meaning to do that in a lot of places) since the top level grid is mostly for some fancier use cases around items moving and stuff

**Comment by robojumper** [comment]
Maybe a bit of a stupid question but can't every Sheet own an ItemPopupContainer and then pass the show/hide callbacks down with Context so that popups are always owned by the correct sheet?

**Comment by bhollis** [comment]
Probably, yeah!

**Comment by robojumper** [comment]
Today I remembered about `allowClickThrough` and that fixes the ClickOutside issue. The one remaining papercut is that while sheets stacking is supported, it doesn't work for Rahools engram focusing, since the item popup that contains the link to engram contents is spawned outside of the sheet hierarchy.

**Comment by bhollis** [comment]
The Rahool engram behavior isn't that annoying. I may fix it with another change anyway.

File: src/app/vendors/single-vendor/SingleVendor.tsx [hunk 1]
```diff
@@ -135,7 +129,7 @@ export default function SingleVendor({ account }: { account: DestinyAccount }) {
   }
 
   return (
-    <div className={clsx(styles.page, 'dim-page', artifactCheck)}>
+    <div className={clsx(artifactCheck)}>
```
Comment (bhollis) [inline src/app/vendors/single-vendor/SingleVendor.tsx:132]: Could remove the `clsx` here

**Comment by robojumper** [comment]
By the way I'm not sure about b8fcaad767c2ad94c6e982f701b658e99cef722c -- if I open the artifact vendor by clicking on the artifact and clicking "preview artifact contents", then the first click on an artifact mod moves the item popup below the sheet (probably because it was already open below it), and the key forces it to be recreated, but I'm not sure why the SearchResults (via selecting an item, hotkey F for search bar, enter to bring search results up, click an item) don't behave that way...

**Review by bhollis — APPROVED** [review]

**Comment by robojumper** [comment]
> but I'm not sure why the SearchResults don't behave that way

hm, actually they do

**Comment by bhollis** [comment]
> selecting an item, hotkey F for search bar, enter to bring search results up, click an item

I can repro that one, and it's just because there's another z-index on SearchResults.

**Review by bhollis — APPROVED** [review]

**Pull request merged**